Mirror `draft` command to current `post` command

ashmaroli opened the pull request:
Adds two features to `draft` command:
  - Define front matter defaults for new drafts using `jekyll_compose.draft_default_front_matter` in `_config.yml`
  - Open new draft in editor based on `EDITOR`  OR `JEKYLL_EDITOR` ENV vars

DirtyF commented:
It works like a charm 👍 

Same on my computer for the tests: 

 - on `master` `script/test` does not open default editor
 - on this branch nano gets fired and prevent the script to exit.

ashmaroli commented:
@DirtyF I've fixed the tests and rebased the branch.
Also, I opted for a separate `draft_default_front_matter` so that users can supply different front matter templates for drafts and posts..

ashmaroli commented:
@DirtyF I've fixed the tests and rebased the branch.
Also, I opted for a separate `draft_default_front_matter` so that users can supply different front matter templates for drafts and posts..

DirtyF commented:
@jekyllbot: merge +minor
